### PR TITLE
Update branding and copyrights

### PR DESCRIPTION
**Pull request by mint-thompson**
Remove MITRE logo from header.

Remove MITRE section from about page. Add HL7 section to about page. Change copyright to HL7. Change maintainers to HL7 community.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ To develop this site locally:
    $ git submodule update --init --recursive
    ```
 2. [Install Hugo](https://gohugo.io/getting-started/installing/)
-3. Add and edit content, refferring to the [Hugo documentation](https://gohugo.io/documentation/)
```
Comment (jafeltra) [inline README.md:30]: Lol wooo! 🎉

**Review by jafeltra — APPROVED** [review]

File: static/hl7-logo-header.png
Comment (cmoesel) [inline static/hl7-logo-header.png:1]: This HL7 logo looks funny next to the larger logos on the About page: 
CleanShot 2024-09-12 at 08 32 51@2x

I found a bigger version that I think would look better: https://fire.ly/wp-content/uploads/2020/11/HL7_int_01.png.webp

Bigger HL7 Logo

**Review by cmoesel — CHANGES_REQUESTED** [review]
This is looking good. I found a couple of things that I think we should fix before merging:

1. The HL7 logo on the About page is too small (see individual comment below).
2. The footer still contains the PRS statement. That should be removed (see screenshot below).

CleanShot 2024-09-12 at 08 38 53@2x

File: static/hl7-logo-header.png
Comment (mint-thompson) [reply static/hl7-logo-header.png:1]: Thank you for finding this bigger logo! It definitely looks much better alongside the other two. I've updated the PR to use it.

**Review by cmoesel — APPROVED** [review]
Looks good. Thank you, Mint!

**Review by jafeltra — APPROVED** [review]

**Pull request merged**